lib/minitest/mock.rb

```diff
@@ -222,8 +227,10 @@ def stub name, val_or_callable, *block_args
 
     yield self
   ensure
-    metaclass.send :undef_method, name
-    metaclass.send :alias_method, name, new_name
-    metaclass.send :undef_method, new_name
+    unless metaclass.nil?
```

Comment:
I'm also not a big fan of this, but can't really think of a better way of doing it. @zenspider, ideas?